src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePlatformProxy.cs

```diff
@@ -1,218 +1,222 @@
+        }
+
+        /// <summary>
+        /// Considered PII. Please ensure that it is hashed.
+        /// </summary>
+        /// <returns>Device identifier</returns>
+        protected override string InternalGetDeviceId()
+        {
+            return Environment.MachineName;
+        }
+
+        public override ILegacyCachePersistence CreateLegacyCachePersistence()
+        {
+            return new InMemoryLegacyCachePersistance();
+        }
+
+        protected override IWebUIFactory CreateWebUiFactory() => new NetCoreWebUIFactory();
+        protected override ICryptographyManager InternalGetCryptographyManager() => new NetCoreCryptographyManager();
+        protected override IPlatformLogger InternalGetPlatformLogger() => new EventSourcePlatformLogger();
+
+        protected override IFeatureFlags CreateFeatureFlags() => new NetCoreFeatureFlags();
+
+        public override Task StartDefaultOsBrowserAsync(string url, bool isBrokerConfigured)
+        {
+            if (DesktopOsHelper.IsWindows())
+            {
+                try
+                {
+                    var psi = new ProcessStartInfo
+                    {
+                        FileName = url,
+                        UseShellExecute = true
+                    };
+                    Process.Start(psi);
+                }
+                catch
+                {
+                    // hack because of this: https://github.com/dotnet/corefx/issues/10361
+                    url = url.Replace("&", "^&");
+                    Process.Start(new ProcessStartInfo("cmd", $"/c start msedge {url}") { CreateNoWindow = true });
+                }
+            }
+            else if (DesktopOsHelper.IsLinux())
             {
                 string sudoUser = Environment.GetEnvironmentVariable("SUDO_USER");
                 if (!string.IsNullOrWhiteSpace(sudoUser))
                 {
-                    throw new MsalClientException(MsalError.LinuxXdgOpen, MsalErrorMessage.LinuxOpenAsSudoNotSupported);
-                }
-
-                try
-                {
-                    ProcessStartInfo psi = null;
-
+                    throw new MsalClientException(
+                        MsalError.LinuxXdgOpen, 
+                        MsalErrorMessage.LinuxOpenAsSudoNotSupported);
+                }
+
+                try
+                {
+                    bool opened = false;
+
                     foreach (string openTool in GetOpenToolsLinux(isBrokerConfigured))
                     {
                         if (TryGetExecutablePath(openTool, out string openToolPath))
                         {
-                            psi = OpenLinuxBrowser(openToolPath, url);
-
+                            OpenLinuxBrowser(openToolPath, url);
+                            opened = true;
                             break;
                         }
                     }
-                    
```

Comment:
whitespace was wrong here, probably because the file was worked on from Linux some time back. Only this and another block has changes.